src/components/CompareResults/TableHeader.tsx

```diff
@@ -97,31 +100,31 @@ function FilterableColumn({
       </Menu>
     </>
   );
-}
+};
 
 type TableHeaderProps = {
   cellsConfiguration: CompareResultsTableConfig[];
-  filters: Map<string, Set<string>>;
-  onToggleFilter: (columnId: string, filters: Set<string>) => unknown;
-  onClearFilter: (columnId: string) => unknown;
+  filters: Map<string, Set<string>>; // Filters for each column, using Map
+  onToggleFilter: (columnId: string, filters: Set<string>) => void;
+  onClearFilter: (columnId: string) => void;
 };
 
-function TableHeader({
+const TableHeader: React.FC<TableHeaderProps> = ({
   cellsConfiguration,
   filters,
   onToggleFilter,
   onClearFilter,
```

Comment:
Good try and thanks for a detailed PR! 

However, the onClearFilter doesn't clear all the filters because you set the default filter for "Confidence" to "High". You'll need to modify onClearFilter to also remove the default filter.

Reply:
Also looks like your changes also introduced a bug to the platform filters

`FAIL src/__tests__/CompareResults/ResultsTable.test.tsx
  ● Results Table › should filter on the Platform column

    expect(received).toEqual(expected) // deep equality

    - Expected  - 1
    + Received  + 0

      Array [
        "a11yr dhtml.html spam opt e10s fission stylo webrender",
        "  - OSX, Improvement, Low",
        "  - Linux, Regression, Medium",
    -   "  - Windows, -, High",
        "  - Windows, -, ",
        "  - Android, Improvement, Low",
      ]

      119 |
      120 |     await screen.findByText('a11yr');
    > 121 |     expect(summarizeVisibleRows()).toEqual([
          |                                    ^
      122 |       'a11yr dhtml.html spam opt e10s fission stylo webrender',
      123 |       '  - OSX, Improvement, Low',
      124 |       '  - Linux, Regression, Medium',

      at Object.toEqual (src/__tests__/CompareResults/ResultsTable.test.tsx:121:36)`